### PR TITLE
💄 #2327 Submitted Data Feedback

### PR DESCRIPTION
**Pull request by demariadaniel**
# Description of changes

- Fixes several inconsistencies with other Clinical/Program pages
- Use snake case URL values
- No Data page displays
- Removes 'gender' and 'submitter_id' fields
- Renders all schema entity fields in order consistent with dictionary

Related Service PR: https://github.com/icgc-argo/argo-clinical/pull/788

https://app.zenhub.com/workspaces/icgc-argo-platform-dk-production-board-5e542d38415f5034e9fed89d/issues/icgc-argo/platform-ui/2327

## Type of Change

- [ ] Bug
- [x] Styling
- [ ] New Feature

## Checklist before requesting review

- Design (select one):
  - [x] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [ ] No changes to UI
- [x] Feature is minimally responsive
- [x] Manual testing of UI feature
- [x] Add copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/submission-system/program-submitted-data/ClinicalEntityDataTable/index.tsx [hunk 1]
```diff
@@ -312,8 +315,8 @@ const ClinicalEntityDataTable = ({
             emptyCompletion;
 
           CoreCompletionFields.forEach((field) => {
-            clinicalRecord[completionColumnHeaders[field]] =
-              completion[field] === null ? 0 : completion[field];
+            const completionField = completionColumnHeaders[field];
+            clinicalRecord[completionField] = completion[field] ? completion[field] : 0;
```
Comment (samrichca) [inline components/pages/submission-system/program-submitted-data/ClinicalEntityDataTable/index.tsx:319]: minor thing, you could do `completion[field] || 0`

File: components/pages/submission-system/program-submitted-data/index.tsx [hunk 2]
```diff
@@ -94,6 +95,7 @@ export default function ProgramSubmittedData() {
       )}
     </VerticalTabs.Item>
   ));
+  const currentEntity: string = reverseEntityNames[selectedClinicalEntityTab];
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply components/pages/submission-system/program-submitted-data/index.tsx:98]: Agreed, this is one thing I felt could be done better, will update

[the remark this answers is not in the document]
Comment (samrichca) [reply components/pages/submission-system/program-submitted-data/index.tsx]: awesome thanks!

**Comment by samrichca** [comment]
Great stuff, LGTM! 🚢

**Review by samrichca — APPROVED** [review]

**Pull request merged**